src/main/java/com/example/onlinestore/service/impl/UserServiceImpl.java

```diff
@@ -75,6 +75,8 @@ public LoginResponse login(LoginRequest request) {
                 return createLoginResponse(request.getUsername());
             } else {
                 logger.warn("管理员密码错误");
+                // 记录失败次数（全局的，不区分用户）
+                recordFailedLogin(request.getUsername());
                 throw new IllegalArgumentException(messageSource.getMessage(
```

Comment:
建议在调用处也加入保护，避免 Redis 异常影响登录失败路径：

```suggestion
                } else {
                    logger.warn("管理员密码错误");
                    try {
                        recordFailedLogin(request.getUsername());
                    } catch (Exception e) {
                        logger.warn("记录失败登录次数异常", e);
                    }
                    throw new IllegalArgumentException(messageSource.getMessage(
                        "error.invalid.credentials", null, LocaleContextHolder.getLocale()));
                }
```


---
*Powered by Qoder*